proof-of-concepts/jit-watcher/main.go

```diff
@@ -0,0 +1,770 @@
+
+	for _, req := range userRequests {
+		resourceCount := w.countResources(req)
+
+		if resourcesToKeep >= resourceCount {
+			resourcesToKeep -= resourceCount
+			w.logDebug("Keeping request %s with %d resources (remaining quota: %d)",
+				req.ID, resourceCount, resourcesToKeep)
+		} else {
+			requestsToLock = append(requestsToLock, req)
+			w.logDebug("Marking request %s for locking (%d resources)", req.ID, resourceCount)
+		}
+	}
+
+	// Lock excess requests
+	if len(requestsToLock) > 0 {
+		w.logInfo("Locking %d requests to enforce resource limit", len(requestsToLock))
+
+		for _, req := range requestsToLock {
+			if w.isRequestLocked(req.ID) {
+				w.logInfo("Request %s already locked", req.ID)
+			} else {
+				reason := fmt.Sprintf("Exceeded maximum approved resources limit (%d)", w.config.MaxResources)
+				resourceNames := make([]string, len(req.Resources))
+				for i, res := range req.Resources {
+					resourceNames[i] = fmt.Sprintf("%s:%s", res.Kind, res.Name)
+				}
+
+				w.logInfo("Locking request %s (created: %s, %d resources: %s)",
+					req.ID, req.Created.Format(time.RFC3339), len(req.Resources), strings.Join(resourceNames, ","))
+
+				if err := w.lockAccessRequest(ctx, req.ID, reason); err != nil {
+					w.logError("Failed to lock request %s: %v", req.ID, err)
+				} else {
+					w.logInfo("Successfully locked request %s for resource limit", req.ID)
+				}
+			}
+		}
+	}
+}
+
+// processAllUsers processes all users and their access requests
+func (w *Watcher) processAllUsers(ctx context.Context) error {
+	w.logInfo("=== Processing all users ===")
+
+	// Get all access requests
+	allRequests, err := w.getAllAccessRequests(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get access requests: %w", err)
+	}
+
+	if len(allRequests) == 0 {
+		w.logInfo("No access requests found")
+		return nil
+	}
+
+	w.logInfo("Found %d total access requests", len(allRequests))
+
+	// Step 1: Process pending requests for auto-approval/denial
+	processedRequests := w.validateAndProcessPendingRequests(ctx, allRequests)
+
+	// Step 2: Group approved requests by user (after auto-approval processing)
+	approvedByUser := w.getApprovedRequestsByUser(processedRequests)
+
+	if len(approvedByUser) == 0 {
+		w.logInfo("No approved requests found")
+		return nil
+	}
+
+	// Step 3: Process each user's approved requests for policy enforcement
+	for user, userRequests := range approvedByUser {
+		w.logInfo("\n=== Processing approved requests for user: %s ===", user)
+
+		// Check environment conflicts first
+		unlockedRequests := w.processEnvironmentConflicts(ctx, userRequests)
+
+		// Check resource limits on remaining unlocked requests
+		w.processResourceLimits(ctx, unlockedRequests)
+	}
+
+	return nil
+}
+
+// Watch starts the monitoring (polling version)
+func (w *Watcher) Watch(ctx context.Context) error {
+	w.logInfo("Starting Teleport JIT Access Request Watcher (Polling Mode)")
+	w.logInfo("Auth Service: %s", w.config.AuthServer)
```

Comment:
Should this be proxy server instead for clarity? I presume it's the overloaded "either auth or proxy" version that we supported back in older configs.